database/blockstate.go

```diff
@@ -60,40 +59,33 @@ func (bs *blockState) execTx(bc *core.BlockChain, config *eth.Config, chainConfi
 		return tmtAbciTypes.ResponseDeliverTx{Code: tmtCode.CodeTypeEncodingError, Log: fmt.Sprintf("Error applying state TX %v", err)}
 	}
 
-	logs := bs.state.GetLogs(tx.Hash())
-
 	bs.txIndex++
 
 	// The slices are allocated in updateBlockState
 	bs.transactions = append(bs.transactions, tx)
 	bs.receipts = append(bs.receipts, receipt)
-	bs.allLogs = append(bs.allLogs, logs...)
 
 	return tmtAbciTypes.ResponseDeliverTx{Code: tmtAbciTypes.CodeTypeOK}
 }
 
 // Persist the eth sate, update the header, make a new block and save it to disk.
 //
-// Returns final application root hash (last block app state hash).
-func (bs *blockState) persist(bc *core.BlockChain, db ethdb.Database) (rootHash common.Hash, err error) {
-	rootHash, err = bs.state.Commit(false)
+// Returns the persisted Block.
+func (bs *blockState) persist(bc *core.BlockChain, db ethdb.Database) (ethTypes.Block, error) {
+	rootHash, err := bs.state.Commit(false)
 	if err != nil {
-		return common.Hash{}, err
+		return ethTypes.Block{}, err
 	}
-
 	bs.header.Root = rootHash
-	for _, log := range bs.allLogs {
-		log.BlockHash = bs.header.Root
-	}
 
 	// Write block to disk
 	block := ethTypes.NewBlock(bs.header, bs.transactions, nil, bs.receipts)
 	_, err = bc.InsertChain([]*ethTypes.Block{block})
 	if err != nil {
-		return common.Hash{}, err
+		return ethTypes.Block{}, err
 	}
 
-	return rootHash, nil
+	return *block, nil
```

Comment:
[MINOR] You are creating a copy of the object. Why didn't you continue the chain of call returning the pointer instead?